### PR TITLE
Bump minimum Python version to 3.9, test with 3.8 - 3.10.

### LINKED ISSUE #1705: Bump minimum python to 3.8 and test 3.8,3.9,3.10
See https://github.com/scverse/anndata/pull/820

### PR DESCRIPTION
**Pull request by martinkim0**
Closes #1705. Changes include:
- Remove `Literal` compatibility type

### CODE REVIEW

**Comment by adamgayoso** [comment]
I realize now that his should be blocked until google colab is upgraded to python 3.8 (and we don't know when that is). We should thus for the short term have another PR that adds python 3.10 to CI without removing anything

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
     timeout-minutes: 20
     strategy:
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        python-version: ["3.8", "3.9", "3.10"]
```
Comment (adamgayoso) [inline .github/workflows/test.yml:18]:
```suggestion
        python-version: ["3.8", "3.9", "3.10", "3.11"]
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ jobs:
     timeout-minutes: 20
     strategy:
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
```
Comment (adamgayoso) [inline .github/workflows/test.yml:18]:
```suggestion
        python-version: ["3.8", "3.9", "3.10"]
```

**Comment by adamgayoso** [comment]
@martinkim0 colab is on python 3.8 now, so we can re prioritize this!

**Comment by martinkim0** [comment]
I'll open a new PR, too many merge conflicts on this one.

**Pull request closed without merging**